torchmetrics/utilities/data.py

```diff
@@ -16,6 +16,16 @@
 import torch
 from torch import Tensor, tensor
 
+from torchmetrics.utilities.imports import _TORCH_GREATER_EQUAL_1_7, _TORCH_GREATER_EQUAL_1_8
+
+if _TORCH_GREATER_EQUAL_1_8:
+    deterministic = torch.are_deterministic_algorithms_enabled
+elif _TORCH_GREATER_EQUAL_1_7:
+    deterministic = torch.is_deterministic
+else:
+    # Assume for older versions that computations need to be deterministic
+    deterministic = lambda: True
```

Comment:
Is this a safe assumption? AFAIK the default behavior is non-deterministic unless you set that explicitly, which for older versions there seems to be no way to do that.